jwst/assign_wcs/assign_wcs_step.py

```diff
@@ -83,5 +90,44 @@ def process(self, input, *args, **kwargs):
             slit_y_range = [self.slit_y_low, self.slit_y_high]
             result = load_wcs(input_model, reference_file_names, slit_y_range)
 
+            if self.sip_approx:
+                if self.sip_degree is not None and self.sip_degree > 6:
+                    raise ValueError(f"SIP polynomial degree of {self.sip_degree} must be <=6")
+                if self.sip_inv_degree is not None and self.sip_inv_degree > 6:
+                    raise ValueError(f"SIP polynomial degree of {self.sip_degree} must be <=6")
+                # check if its imaging mode
+                if result.meta.exposure.type.lower() in IMAGING_TYPES:
+
+                    # fit sip approx., degree is chosen by best fit
+                    try:
+                        fit_sip_hdr = result.meta.wcs.to_fits_sip(None, self.sip_max_pix_error,
+                                      self.sip_degree, self.sip_max_inv_pix_error, self.sip_inv_degree,
+                                      self.sip_npoints)
+                    except ValueError:
+                        return result
+
+
+                    # if a polynomia was fit (not just linear terms), but the
+                    # best fit > 6th degree, fit a 6th degree instead.
+                    # and give up if that doesn't work.
+                    if 'SIP' in fit_sip_hdr['ctype1'] and fit_sip_hdr['a_order'] > 6:
```

Comment:
I think this still does not cover all situations. For example, what if forward polynomial degree is 6 (OK) but inverse is 7? This logic would not detect this situation.

I would do it this way:
```
                    if 'SIP' in fit_sip_hdr['ctype1'] and (fit_sip_hdr['a_order'] > 6 or fit_sip_hdr['ap_order'] > 6):
                        fwd_deg = min(fit_sip_hdr['a_order'], 6)
                        inv_deg = min(fit_sip_hdr['ap_order'], 6)
                        result.meta.wcs.to_fits_sip(None, self.sip_max_pix_error,
                                            fwd_deg, self.sip_max_inv_pix_error, inv_deg, self.sip_npoints)
```